source/common/network/address_impl.cc

```diff
@@ -31,6 +31,19 @@ const SocketInterface* sockInterfaceOrDefault(const SocketInterface* sock_interf
   return sock_interface == nullptr ? &SocketInterfaceSingleton::get() : sock_interface;
 }
 
+void throwOnError(StatusOr<InstanceConstSharedPtr> address) {
+  ASSERT(Thread::MainThread::isMainThread());
```

Comment:
move this ASSERT into the absl::Status version of throwOnError below, and just call it rather than duplicating its impl.

Reply:
Can't move the assertion there, it is called by several filters.

Reply:
You can't just leave it not moved; it obviously should be moved :)

You could leave comments and a TODO explaining why it isn't moved yet.